src/builder/Actions.sol

```diff
@@ -84,25 +96,130 @@ library Actions {
         uint256 amount;
         uint256 price;
         address token;
+        string assetSymbol;
         uint256 chainId;
         address recipient;
         uint256 destinationChainId;
         string bridgeType;
     }
 
-    function bridgeAsset(BridgeAsset memory bridge, PaymentInfo.Payment memory payment, bool useQuoteCall)
+    function constructBridgeOperations(
+        BridgeOperationInfo memory bridgeInfo,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) internal view returns (IQuarkWallet.QuarkOperation[] memory, Action[] memory) {
+        /*
+         * at most one bridge operation per non-destination chain,
+         * and at most one transferIntent operation on the destination chain.
+         *
+         * therefore the upper bound is chainAccountsList.length.
+         */
+        uint256 actionIndex = 0;
+        Action[] memory actions = new Action[](chainAccountsList.length);
+        IQuarkWallet.QuarkOperation[] memory quarkOperations =
+            new IQuarkWallet.QuarkOperation[](chainAccountsList.length);
+
+        // Note: Assumes that the asset uses the same # of decimals on each chain
+        uint256 balanceOnDstChain =
+            Accounts.getBalanceOnChain(bridgeInfo.assetSymbol, bridgeInfo.dstChainId, chainAccountsList);
+        uint256 amountLeftToBridge = bridgeInfo.amountNeededOnDst - balanceOnDstChain;
+        uint256 bridgeActionCount = 0;
+        // TODO: bridge routing logic (which bridge to prioritize, how many bridges?)
+        // Iterate chainAccountList and find up to 2 chains that can provide enough fund
```

Comment:
Is this comment still true that it runs upto 2 chains?

Reply:
Good catch, not anymore. Updated the comment